backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -1140,38 +1137,34 @@ public void deleteBranchByCommit(Session session, Long repoId, String commitHash
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
       try {
-        WorkspaceDTO workspaceDTO = null;
-        List<String> accessibleResourceIds =
-            roleService.getAccessibleResourceIds(
-                null,
-                new CollaboratorUser(authService, currentLoginUserInfo),
-                ModelDBServiceResourceTypes.REPOSITORY,
-                request.getRepoIdsList().stream()
-                    .map(String::valueOf)
-                    .collect(Collectors.toList()));
+        Set<String> accessibleResourceIds =
+            new HashSet<>(
+                roleService.getAccessibleResourceIds(
+                    null,
+                    new CollaboratorUser(authService, currentLoginUserInfo),
+                    ModelDBServiceResourceTypes.REPOSITORY,
+                    request.getRepoIdsList().stream()
+                        .map(String::valueOf)
+                        .collect(Collectors.toList())));
 
         String workspaceName = request.getWorkspaceName();
-        if (workspaceName != null
-            && !workspaceName.isEmpty()
+        if (!workspaceName.isEmpty()
             && workspaceName.equals(authService.getUsernameFromUserInfo(currentLoginUserInfo))) {
-          LOGGER.debug("Workspace found and match with current login username");
+          List<GetResourcesResponseItem> accessibleAllWorkspaceItems =
+              roleService.getResourceItems(
+                  null, accessibleResourceIds, ModelDBServiceResourceTypes.REPOSITORY);
           accessibleResourceIds =
-              roleService.getSelfDirectlyAllowedResources(
-                  ModelDBServiceResourceTypes.REPOSITORY, ModelDBServiceActions.READ);
-          LOGGER.debug(
-              "Self directly accessible Repository Ids found, size {}",
-              accessibleResourceIds.size());
-          if (request.getRepoIdsList() != null && !request.getRepoIdsList().isEmpty()) {
-            accessibleResourceIds.retainAll(
-                request.getRepoIdsList().stream()
-                    .map(String::valueOf)
-                    .collect(Collectors.toList()));
-          }
+              accessibleAllWorkspaceItems.stream()
+                  .map(GetResourcesResponseItem::getResourceId)
+                  .collect(Collectors.toSet());
         } else {
-          LOGGER.debug("Workspace not found or not match with current login username");
-          workspaceDTO =
-              roleService.getWorkspaceDTOByWorkspaceName(
-                  currentLoginUserInfo, request.getWorkspaceName());
+          accessibleResourceIds =
+              ModelDBUtils.filterWorkspaceOnlyAccessibleIds(
```

Comment:
No, from `ModelDBUtils.filterWorkspaceOnlyAccessibleIds(` method i am calling the UAC with accessibleResourceIdsWithCollaborator and workspace so the UAC return ids with filtering only my workspace ids.

Reply:
internally it is used  `roleService.getResourceItems` with the workspace.

Reply:
I believe you are overfetching since you called `getAccessibleResourceIds`, which returns everything regardless of what workspace they belong to, and then call `filterWorkspaceOnlyAccessibleIds` which is filtering, as the name suggests. What am I missing here?

Reply:
Thread: https://vertaai.slack.com/archives/C01JVHUHKR6/p1610415593088700